Added a text field that imports user-inputted filter directly to feature xml file

gian-benito opened the pull request:
Changes: 
- Added a filter textfield
- Textfield contents are validated by parsing as an OSGi filter
- Textfield contents are also verified to have valid syntax
- User input is called by setFilter into an XML write function, which directly imports it into the corresponding XML file. 
- tested for multiple features and existing Eclipse PDE features
<img width="1070" height="656" alt="image" src="https://github.com/user-attachments/assets/3b907fb5-281c-4e76-b675-0e745b3107e5" />

<img width="1067" height="674" alt="image" src="https://github.com/user-attachments/assets/f4f76f93-7f07-4673-8f51-66f93264fbb0" />

@KN0987 @BAlgarra
Related to #1808

HannesWell commented on ui/org.eclipse.pde.core/src/org/eclipse/pde/core/plugin/IPluginReference.java:
```diff
@@ -70,4 +70,15 @@ public interface IPluginReference extends IIdentifiable, IMatchRules {
 	 * @throws CoreException if the model is not editable
 	 */
 	void setVersion(String version) throws CoreException;
+
+	void setFilter(String filter) throws CoreException;
+
+	/**
+	 * Returns the filter string for this plugin reference.
+	 *
+	 * @return the filter string, or null if not set
+	 * @since 3.21
+	 */
+	String getFilter();
+
```
Why did you add these methods this high in the type hierarchy? Wouldn't it be sufficient to add in `IPluginImport`?
Furthermore the set method is missing the `@since` tag and proper documentation. Then an API filter should not be necessary.

HannesWell commented on ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/plugin/PluginImport.java:
```diff
@@ -312,4 +312,16 @@ public void writeDelimeter(PrintWriter writer) {
+
+	@Override
+	public String getFilter() {
+		// TODO Auto-generated method stub
+		return null;
```
This get and set method have to be implemented. But this time the value should be stored in a filed, as you currently do it in `PluginImportNode`.

HannesWell commented on ui/org.eclipse.pde.core/text/org/eclipse/pde/internal/core/text/plugin/PluginImportNode.java:
```diff
@@ -20,6 +20,18 @@
+
+	@Override
+	public void setFilter(String filter) {
+		this.fFilter = filter;
```
As this represents the PluginImport node of an XML file and looking at the other methods, the implementation seems to have to look more like
```suggestion
		setXMLAttribute(P_FILTER, filter);
```
with a new constant `String P_FILTER = "filter"; //$NON-NLS-1$` added in `IPluginImport`.

HannesWell commented on ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/MatchSection.java:
```diff
@@ -134,6 +139,28 @@ public void textDirty(FormEntry text) {
+					@Override
+					public void textValueChanged(FormEntry text) {
+						applyFilter(text.getValue());
+
+
```
```suggestion
```

HannesWell commented on ui/org.eclipse.pde.ui/src/org/eclipse/pde/ui/templates/PluginReference.java:
```diff
@@ -114,4 +114,16 @@ public void setId(String id) throws CoreException {
 		this.id = id;
 	}
 
+	@Override
+	public void setFilter(String filter) throws CoreException {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public String getFilter() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
```
This is actually not necessary isn't it?

```suggestion
```

HannesWell reviewed (changes requested):
Thank you for this contribution to enhance Eclipse PDE.

For this first round I have multiple remarks below.
After you have addressed them, please squash all your changes/commits into one with a message covering the overall change. Also make sure you have rebased on top of the laster master of this repository.
Then force-push that one commit to the branch of this PR, so that this PR contains a single commit. Please DO NOT create a new PR.
If you need assistance with that, please ask your Mentor from CodeDays for assistance.

gian-benito commented:
Hi @HannesWell. Just to follow on up this pull request, we spoke with a CodeDay mentor and they advised us to open a new PR since we won't be able to reopen it on our side. Is there any possibility you can re-open it on your end? We have also addressed the issues and squashed our commits into one.

HannesWell commented:
> we spoke with a CodeDay mentor and they advised us to open a new PR since we won't be able to reopen it on our side. Is there any possibility you can re-open it on your end? We have also addressed the issues and squashed our commits into one.

I can't reopen it either. Under some circumstances (branch still exists and points to the very same commit, with same SHA) one can reopen PRs, but in this case these conditions don't seem to be met. If it's possible, then you should have the `Reopen ...` button enabled:

<img width="800" src="https://github.com/user-attachments/assets/6c43f565-3b7f-4aeb-96fb-06f1b270198f" />

But if that doesn't work, please create a new PR and mention this one in the comment.
